test/functional/ui/statuscolumn_spec.lua

```diff
@@ -589,6 +590,27 @@ describe('statuscolumn', function()
         meths.input_mouse('left', 'press', '', 0, 7, 7)
         eq('0 1 l 11', eval("g:testvar"))
       end)
+
+      it('selecting popupmenu does not drag mouse', function()
+        screen:try_resize(screen._width, 2)
+        meths.set_option_value('statuscolumn', '%0@MyClickFunc@%=%l%T', {})
+        exec([[
+          function! MyClickFunc(minwid, clicks, button, mods)
+            let g:testvar = printf("%d %d %s %d", a:minwid, a:clicks, a:button, getmousepos().line)
+            menu PopupStc.Echo <cmd>echo g:testvar<CR>
+            popup PopupStc
+          endfunction
+        ]])
+        meths.input_mouse('left', 'press', '', 0, 0, 0)
+        poke_eventloop()
+        meths.input_mouse('left', 'press', '', 0, 1, 5)
+        poke_eventloop()
+        meths.input_mouse('left', 'release', '', 0, 1, 5)
```

Comment:
Sorry, actually `poke_eventloop()` won't work here. A `screen:expect()` is needed between the first and second click, and nothing is needed between the click and the release.